src/workerd/api/streams/readable.h

```diff
@@ -360,13 +360,22 @@ class ReadableStream: public jsg::Object {
   // disturbed and the DeferredProxy returned will take over ownership of the internal
   // state of the readable.
 
+  kj::Maybe<jsg::PromiseResolverPair<void>> eofResolverPair;
```

Comment:
It's generally considered bad form to declare public member variables on C++ classes. In this case in particular, we don't want callers to be able to invoke the resolver, we only want them to be able to wait on the promise.

Can we make this private, and replace `initEofResolverPair` with:

```
jsg::Promise<void> onEof(jsg::Lock& js);
```

This function would initialize the pair and return the promise in a single shot. We could document that it cannot be called more than once.